backend/x11/backend.c

```diff
@@ -609,35 +615,17 @@ struct wlr_backend *wlr_x11_backend_create(struct wl_display *display,
 		x11->drm_fd = query_dri3_drm_fd(x11);
 		if (x11->drm_fd < 0) {
 			wlr_log(WLR_ERROR, "Failed to query DRI3 DRM FD");
-			goto error_event;
-		}
-
-		char *drm_name = drmGetDeviceNameFromFd2(x11->drm_fd);
-		wlr_log(WLR_DEBUG, "Using DRM node %s", drm_name);
-		free(drm_name);
-
-		int drm_fd = fcntl(x11->drm_fd, F_DUPFD_CLOEXEC, 0);
-		if (drm_fd < 0) {
-			wlr_log(WLR_ERROR, "fcntl(F_DUPFD_CLOEXEC) failed");
-			goto error_event;
-		}
-
-		struct wlr_gbm_allocator *gbm_alloc = wlr_gbm_allocator_create(drm_fd);
-		if (gbm_alloc == NULL) {
-			wlr_log(WLR_ERROR, "Failed to create GBM allocator");
-			close(drm_fd);
-			goto error_event;
+			if (x11->have_shm) {
```

Comment:
TBH it's probably not worth it to fallback to shm when `query_dri3_drm_fd` fails. On a GPU-less system, `have_dri3` will be false.